File: Backend/Services/Tickets/TicketingSystem.Tickets.Domain/Models/Ticket.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace TicketingSystem.Tickets.Domain.Models
{
    public class Ticket
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        public TicketType TicketType { get; set; } = new TicketType();
        public User UserWhoCreated { get; set; } = new User();
        public TicketServiceType ServiceType { get; set; } = new TicketServiceType();
        public string Subject { get; set; } = "";
        public string Description { get; set; } = "";
        public TicketStatus Status { get; set; } = new TicketStatus();
        public DateTime Opened { get; set; }
        public string CustomerName { get; set; }
        public DateTime Closed { get; set; }
        public TicketPriority Priority { get; set; }
    }
}

File: Frontend/ticket-system/src/app/tickets/ticket.ts
import { TicketPriority } from "./TicketPriority";
import { TicketStatus } from "./TicketStatus";
import { TicketServiceType } from "./TicketServiceType";
import { TicketType } from "./TicketType";

export type Ticket = {
    id: number,
    ticketType: TicketType,
    serviceType:TicketServiceType,
    subject: string,
    description:string,
    status: TicketStatus,
    opened: Date,
    customerName:string,
    closed: Date,
    priority: TicketPriority,
}